GeneralsMD/Code/GameEngineDevice/Source/StdDevice/Common/StdLocalFileSystem.cpp

```diff
@@ -32,43 +32,32 @@
 #include "StdDevice/Common/StdLocalFileSystem.h"
 #include "StdDevice/Common/StdLocalFile.h"
 
-#include <filesystem>
-
 StdLocalFileSystem::StdLocalFileSystem() : LocalFileSystem() 
 {
 }
 
 StdLocalFileSystem::~StdLocalFileSystem() {
 }
 
-//DECLARE_PERF_TIMER(StdLocalFileSystem_openFile)
-File * StdLocalFileSystem::openFile(const Char *filename, Int access /* = 0 */) 
+std::filesystem::path StdLocalFileSystem::fixFilenameFromWindowsPath(const Char *filename, Int access) const
 {
-	//USE_PERF_TIMER(StdLocalFileSystem_openFile)
-	StdLocalFile *file = newInstance( StdLocalFile );	
-
-	// sanity check
-	if (strlen(filename) <= 0) {
-		file->deleteInstance();
-		return NULL;
-	}
-
 	std::string fixedFilename(filename);
 
-#ifndef _WIN32
+	#ifndef _WIN32
 	// Replace backslashes with forward slashes on unix
 	std::replace(fixedFilename.begin(), fixedFilename.end(), '\\', '/');
-#endif
+	#endif
 
 	// Convert the filename to a std::filesystem::path and pass that
 	// make_preferred does not convert from Windows to Linux (see https://en.cppreference.com/w/cpp/filesystem/path/make_preferred)
 	std::filesystem::path path(fixedFilename);
 	path = path.make_preferred();
 
-#ifndef _WIN32
+	#ifndef _WIN32
 	// check if the file exists to see if fixup is required
 	// if it's not found try to match disregarding case sensitivity
-	if (!std::filesystem::exists(path)) {
+	// For cases where a write is happening, we should check if the parent path exists, if so, let it through, since the file may not exist yet.
+	if (!std::filesystem::exists(path) && !std::filesystem::exists(path.parent_path())) {
```

Comment:
Updating this check to check for parent path was also needed for map thumbnails, since it's ok if the full path does not exist for a write.